Exercise 1 BFile/myFile-rt/src/com/tridiumuniversity/myFile/BFile.java

```diff
@@ -0,0 +1,214 @@
+  public int getSize() { return getInt(size); }
+  
+  /**
+   * Set the {@code size} property.
+   * @see #size
+   */
+  public void setSize(int v) { setInt(size, v, null); }
+
+////////////////////////////////////////////////////////////////
+// Property "createdDate"
+////////////////////////////////////////////////////////////////
+  
+  /**
+   * Slot for the {@code createdDate} property.
+   * @see #getCreatedDate
+   * @see #setCreatedDate
+   */
+  public static final Property createdDate = newProperty(0, BAbsTime.make(), null);
+  
+  /**
+   * Get the {@code createdDate} property.
+   * @see #createdDate
+   */
+  public BAbsTime getCreatedDate() { return (BAbsTime)get(createdDate); }
+  
+  /**
+   * Set the {@code createdDate} property.
+   * @see #createdDate
+   */
+  public void setCreatedDate(BAbsTime v) { set(createdDate, v, null); }
+
+////////////////////////////////////////////////////////////////
+// Property "modifiedDate"
+////////////////////////////////////////////////////////////////
+  
+  /**
+   * Slot for the {@code modifiedDate} property.
+   * @see #getModifiedDate
+   * @see #setModifiedDate
+   */
+  public static final Property modifiedDate = newProperty(0, BAbsTime.NULL, null);
+  
+  /**
+   * Get the {@code modifiedDate} property.
+   * @see #modifiedDate
+   */
+  public BAbsTime getModifiedDate() { return (BAbsTime)get(modifiedDate); }
+  
+  /**
+   * Set the {@code modifiedDate} property.
+   * @see #modifiedDate
+   */
+  public void setModifiedDate(BAbsTime v) { set(modifiedDate, v, null); }
+
+////////////////////////////////////////////////////////////////
+// Action "touch"
+////////////////////////////////////////////////////////////////
+  
+  /**
+   * Slot for the {@code touch} action.
+   * @see #touch()
+   */
+  public static final Action touch = newAction(0, null);
+  
+  /**
+   * Invoke the {@code touch} action.
+   * @see #touch
+   */
+  public void touch() { invoke(touch, null, null); }
+
+////////////////////////////////////////////////////////////////
+// Action "print"
+////////////////////////////////////////////////////////////////
+  
+  /**
+   * Slot for the {@code print} action.
+   * @see #print()
+   */
+  public static final Action print = newAction(0, null);
+  
+  /**
+   * Invoke the {@code print} action.
+   * @see #print
+   */
+  public void print() { invoke(print, null, null); }
+
+////////////////////////////////////////////////////////////////
+// Type
+////////////////////////////////////////////////////////////////
+  
+  @Override
+  public Type getType() { return TYPE; }
+  public static final Type TYPE = Sys.loadType(BFile.class);
+
+/*+ ------------ END BAJA AUTO GENERATED CODE -------------- +*/
+
+  public void doTouch(){
+    setModifiedDate(BAbsTime.make());
+  }
+
+  public void doPrint(){
+    System.out.println("fileName : " + getFileName());
+    System.out.println("path : " + getPath());
+    System.out.println("size : " + getSize() + " bytes");

```

Comment:
I like that you specified bytes here! This exercise didn't specify that you do this, but you can also add units to properties as facets! That would look something like the following:

```
@NiagaraProperty(
  name = "size",
  type = "BInteger",
  defaultValue = "BInteger.DEFAULT",
  facets = @Facet(name = "BFacets.UNITS", value = "BUnit.getUnit(\"byte\")")
)
```